mmrazor/models/distillers/ofd_distiller.py

```diff
@@ -32,20 +32,21 @@ def init_ofd_connectors(self, teacher: nn.Module) -> None:
             if isinstance(self.distill_losses[loss_key], OFDLoss):
                 for _input_keys, _input_mapping in loss_forward_mapping.items(
                 ):
-                    recorder_mgn = self.student_recorders if _input_mapping[
-                        'from_student'] else self.teacher_recorders
-                    recorder = recorder_mgn.get_recorder(
-                        _input_mapping['recorder'])
-                    module_key = recorder.source
-                    bn_module = attrgetter(module_key)(teacher)
-
-                    assert isinstance(
-                        bn_module, (nn.BatchNorm2d, nn.SyncBatchNorm)), (
-                            'Overhaul distillation only support connection on '
-                            'layers: [`BatchNorm2d`, `SyncBatchNorm`]')
-
                     if 'connector' in _input_mapping and not _input_mapping[
                             'from_student']:
+
+                        recorder_mgn = self.student_recorders \
+                                if _input_mapping['from_student'] \
+                                else self.teacher_recorders
```

Comment:
The above judgment (Line 35-36) has limited `not _input_mapping['from_student']`, here should no longer appear `_input_mapping['from_student'] == True`.